eval/src/vespa/eval/instruction/join_with_number_function.cpp

```diff
@@ -64,22 +80,27 @@ JoinWithNumberFunction::~JoinWithNumberFunction() = default;
 bool
 JoinWithNumberFunction::inplace() const {
     if (_primary == Primary::LHS) {
-        return lhs().result_is_mutable();
+        return lhs().result_type() == result_type()
+            && lhs().result_is_mutable();
     } else {
-        return rhs().result_is_mutable();
+        return rhs().result_type() == result_type()
+            && rhs().result_is_mutable();
     }
 }
 
-using MyTypify = TypifyValue<TypifyCellType,vespalib::TypifyBool,operation::TypifyOp2>;
+using MyTypify = TypifyValue<TypifyCellMeta,vespalib::TypifyBool,operation::TypifyOp2>;
 
 InterpretedFunction::Instruction
-JoinWithNumberFunction::compile_self(const ValueBuilderFactory &, Stash &) const
+JoinWithNumberFunction::compile_self(const ValueBuilderFactory &, Stash &stash) const
 {
-    auto op = typify_invoke<4,MyTypify,SelectJoinWithNumberOp>(result_type().cell_type(),
+    const auto &param = stash.create<JoinWithNumberParam>(result_type(), _function);
+    auto input_type = (_primary == Primary::LHS) ? lhs().result_type() : rhs().result_type();
+    assert(result_type() == input_type.map());
```

Comment:
this is a convenient truth, but not really for us to decide. prefer checking join with ValueType::double_type() instead.